app/jobs/business_type_update_job.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+class BusinessTypeUpdateJob < ApplicationJob
+    queue_as :default
+  
+    def perform
+        logger = Rails.logger
+        logger.info 'performing BusinessTypeUpdateJob'
+
+        business_type_mapping = Hash[*
+            BusinessType.pluck(:slug, :id).flatten]
+
+        slug_by_id = Hash[*
+            BusinessType.pluck(:id, :slug).flatten]
+
+        Business.find_each do |business|
+            unless business.gmap_id.nil?
+                business_type_id = ApplicationController.helpers.maybe_get_business_type(business_type_mapping, business.gmap_id)
```

Comment:
Should be enough to call `helpers.maybe_get_business_type` instead of `ApplicationController.helpers.maybe_get_business_type`

Reply:
Does this really work? Helpers are normally only available in controllers and templates, as they are normally used for formatting data for output.

The `maybe_get_business_type` method could be moved to a [concern](https://medium.com/@carlescliment/about-rails-concerns-a6b2f1776d7d) instead. That could be called `BusinessTypeMapping`. (There is already a directory for this at *app/models/concerns*.)

Or, if you want to do more OOP, just make that a class and instantiate it where needed (e.g. `BusinessTypeMapping.new(business_type_mapping).map(business.gmap_id)`. Or not even give the mapping to the constructor but load that in the new class as well.

Reply:
I think it only worked when qualifying the helpers with the ApplicationController class. Maybe because the usage is in jobs.
I am not so sure whether a Concern would be appropriate here
https://medium.com/@carlescliment/about-rails-concerns-a6b2f1776d7d

Reply:
Yes, those are good arguments! So I think the explicit approach with a separate class makes most sense here.